src/Arcus.Scripting.LogicApps/Scripts/Resubmit-FailedAzLogicAppRuns.ps1

```diff
@@ -0,0 +1,50 @@
+param(
+    [Parameter(Mandatory = $true)][string] $ResourceGroupName = $(throw "Name of the resource group is required"),
+    [Parameter(Mandatory = $true)][string] $LogicAppName = $(throw "Name of the logic app is required"),
+    [Parameter(Mandatory = $true)][datetime] $StartTime = $(throw "Start time is required"),
+    [Parameter(Mandatory = $false)][datetime] $EndTime
+)
+
+try{
+    if ($EndTime) {
+        $runs = Get-AzLogicAppRunHistory -ResourceGroupName $ResourceGroupName -Name $LogicAppName | 
+            Where-Object {$_.Status -eq 'Failed' -and $_.StartTime -ge $StartTime -and $_.EndTime -le $EndTime}
+    } else {
+        $runs = Get-AzLogicAppRunHistory -ResourceGroupName $ResourceGroupName -Name $LogicAppName | 
+            Where-Object {$_.Status -eq 'Failed' -and $_.StartTime -ge $StartTime}
+    }
```

Comment:
For the sake of DRY, I wonder if it wouldn't be a better option to implement this like this:

(Not checked, so probably no valid PS, but you'll catch the idea)

```
if ($EndTime -eq $null) {
   $EndTime = Get-Date()
}

$runs = Get-AzLogicAppRunHistory - .... ... $_.StartTime -ge $StartTime -and $_.EndTime -le $EndTime ...
```
}
```

Reply:
The thing is I would like to prevent the usage of `-and $_.EndTime -le $EndTime` in the `Where-Object` statement if the EndTime is not provided, just to make sure this does not filter out anything. 
I don't really know of any way to this without having two separate `Get-AzLogicAppRunHistory` though...

Reply:
After offline discussion we decided to keep it as is for now, we can always refactor later on.